cli/cmd/testdata/install_custom_domain.golden

```diff
@@ -907,6 +909,7 @@ spec:
         - mountPath: /var/run/secrets/tokens
           name: linkerd-identity-token
       initContainers:
+      
```

Comment:
Any idea what is causing these empty lines?

Reply:
They don't seem to be introduced by this PR 🤔 I've looked around and I have a bit of an idea about what's going on.

On latest main if I build the CNI and print out the install manifests:

```shell
:; git branch -v
  dependabot/cargo/clap-4.0.19               c430080e9 build(deps): bump clap from 4.0.18 to 4.0.19
  eliza/issue-9671                           9b777750b update goldenfiles
* main                                       4ea8ab21d edge-22.11.3 change notes (#9884)
  matei/add-init-priv-escalation             3c0635b7f PR feedback


:; bin/linkerd install  --ignore-cluster| rg 'initContainer' -A 2
      initContainers:

      - args:
--
      initContainers:

      - args:
--
      initContainers:
      - args:
        - --incoming-proxy-port

```

Culprit seems to be this line:
https://github.com/linkerd/linkerd2/blob/4ea8ab21dc3dfc72542eeb05f895f12f4147f7ac/charts/linkerd-control-plane/templates/destination.yaml#L313
(and the equivalent line for `identity`, injector doesn't seem to suffer from this).

To fix, I've added `-` at the start of the if for `destination` and `identity`.

Result:

```shell
:; git branch
  dependabot/cargo/clap-4.0.19
  eliza/issue-9671
  main
* matei/add-init-priv-escalation

:; bin/linkerd install  --ignore-cluster| rg 'initContainer' -A 2
      initContainers:
      - args:
        - --incoming-proxy-port
--
      initContainers:
      - args:
        - --incoming-proxy-port
--
      initContainers:
      - args:
        - --incoming-proxy-port
```

Reply:
The templates don't seem to want to update themselves through `go test ./... -update` though